src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -243,17 +245,19 @@ internal static bool HasSufficientUniquenessFactor(HashSet<string> set, ReadOnly
 
         internal readonly struct AnalysisResults
         {
-            public AnalysisResults(bool ignoreCase, bool allAsciiIfIgnoreCase, int hashIndex, int hashCount, int minLength, int maxLength)
+            public AnalysisResults(bool ignoreCaseForHash, bool ignoreCaseForEquals, bool allAsciiIfIgnoreCase, int hashIndex, int hashCount, int minLength, int maxLength)
             {
-                IgnoreCase = ignoreCase;
+                IgnoreCaseForHash = ignoreCaseForHash;
+                IgnoreCaseForEquals = ignoreCaseForEquals;
```

Comment:
Nit: case insensitivity for equality is just case sensitivity. 
```suggestion
                IgnoreCase = ignoreCase;
```